src/gui/vector/qgsstackeddiagramproperties.cpp

```diff
@@ -183,7 +183,7 @@ void QgsStackedDiagramProperties::syncToLayer()
     mStackedDiagramSpacingSpinBox->setValue( settingList.at( 0 ).stackedDiagramSpacing() );
     mStackedDiagramSpacingUnitComboBox->setUnit( settingList.at( 0 ).stackedDiagramSpacingUnit() );
 
-    if ( dr->rendererName() == QgsStackedDiagram::DIAGRAM_NAME_STACKED )
+    if ( dr->rendererName() == QLatin1String( "Stacked" ) )
     {
       const QgsStackedDiagramRenderer *stackedDiagramRenderer = static_cast< const QgsStackedDiagramRenderer * >( dr );
```

Comment:
May I suggest the below approach which seems to be to be less fragile ?
```suggestion
    if ( const QgsStackedDiagramRenderer *stackedDiagramRenderer = dynamic_cast< const QgsStackedDiagramRenderer * >( dr ) )
    {
```

or alternatively adding a

const QString QgsStackedDiagramRenderer::RENDERED_NAME = QStringLiteral( "Stacked" );

and use it in the QgsStackedDiagramRenderer::rendererName() method implementation, and in above qgsdiagramwidget.cpp

Those are just suggestions (I tend to hate comparing against string literals all over the place)

Reply:
Thank you, let's pick the second option.
Done in https://github.com/qgis/QGIS/pull/59015/commits/bb8c42ebc3076419527c378b4463f5660e044aeb